pandevice/panorama.py

```diff
@@ -78,23 +79,109 @@ def _setup(self):
 
         self._params = tuple(params)
 
+    @property
+    def vsys(self):
+        return self.name
+
     def devicegroup(self):
         return self
 
     def xpath_vsys(self):
-        if self.name == "shared" or self.name is None:
-            return "/config/shared"
-        else:
-            return "/config/devices/entry[@name='localhost.localdomain']/device-group/entry[@name='%s']" % self.name
+        return self.xpath()
 
-    def _build_xpath(self, root, vsys):
-        if (self.name == "shared" or self.name is None) and root == Root.VSYS:
-            # This is a Vsys object in shared Panorama scope, so proceed normally
-            return super(DeviceGroup, self)._build_xpath(root,
-                self.name or 'shared')
-        else:
-            # This is a member of the device-group, so override to use DeviceGroup root
-            return super(DeviceGroup, self)._build_xpath(self.ROOT, self.name)
+
+class Template(VersionedPanObject):
+    """A panorama template.
+
+    Args:
+        name: Template name
+        description: Description
+        devices (str/list): The list of serial numbers in this template
+        default_vsys: The default vsys in case of a single vsys firewall
+        multi_vsys (bool): (6.1 and lower) Multi virtual systems boolean
+        mode: (6.1 and lower) Can be fips, cc, or normal (default: normal)
+        vpn_disable_mode (bool): (6.1 and lower) VPN disable mode
+
+    """
+    ROOT = Root.DEVICE
+    SUFFIX = ENTRY
+    CHILDTYPES = (
+        "device.Vsys",
+    )
+
```

Comment:
What about all the non-vsys objects?  They should probably be allowed children of Template. For example, ManagementProfiles would be a potential child of Template.